watchtower/wtdb/tower_db.go

```diff
@@ -98,23 +98,36 @@ func OpenTowerDB(dbPath string) (*TowerDB, error) {
 	}
 
 	if firstInit {
-		// The file didn't exist before, initialize a fresh database
-		// with the latest version and bucket structure.
-		err = towerDB.db.Update(initTowerDBBuckets)
+		// If the database has not yet been created, we'll initialize
+		// the database version with the latest known version.
+		err = towerDB.db.Update(func(tx *bbolt.Tx) error {
+			return initDBVersion(tx, getLatestDBVersion(dbVersions))
+		})
 		if err != nil {
 			bdb.Close()
 			return nil, err
 		}
 	} else {
-		// Otherwise, ensure that any migrations are applied before
-		// returning the opened database.
+		// Otherwise, ensure that any migrations are applied to ensure
+		// the data is in the format expected by the latest version.
```

Comment:
If the database file was created, we crash before the if statement body above is executed, then this will fail since there's no metadata bucket.

Reply:
gotcha, i see what you're saying now. `firstInit` should be `file DNE || metadataBkt DNE`

Reply:
should be fixed by fe64b8c